pkgs/by-name/gp/gpt4all/package.nix

```diff
@@ -47,12 +47,9 @@ stdenv.mkDerivation (finalAttrs: {
   ] ++ lib.optionals cudaSupport (
       with cudaPackages;
       [
-        cuda_cccl.dev
-        cuda_cudart.dev
-        cuda_cudart.lib
-        cuda_cudart.static
-        libcublas.dev
-        libcublas.lib
+        cuda_cccl
+        cuda_cudart
+        libcublas
```

Comment:
I'm confused, quite possibly out of the loop on this: Why are we going back to the coarse outputs? Didn't we recently invest effort into splitting these outputs?

Reply:
We invested an effort into splitting the outputs, but then we had a ton of packages use the symlink farm from `$out` which rendered the effort futile: a store path that references `libcudart.so` through the symlink farm ends up pulling the `.a`s in.

What you see in the highlighted diff instead is inputs that `mkDerivation` automatically resolves into the (small) `dev` outputs which _propagate_ all other outputs. The build closure expands, huge (it'll include static archives). The runtime closure, however, retains references to `.lib`s but not to `.dev`s and `.static`s. My expectation: runtime closures shrink.

Reply:
> which rendered the effort futile: a store path that references libcudart.so through the symlink farm ends up pulling the .as in.

This behavior only happens when the symlink farm includes static stuff though, right? Eg if I do `symlinkJoin { cudaPackages.foo.lib }` it will not pull in any static archives IIUC?

> The build closure expands, huge (it'll include static archives). The runtime closure, however, retains references to .libs but not to .devs and .statics. My expectation: runtime closures shrink.

I see. Smaller runtime closures are always welcome.

Reply:
We could still trade some complexity for smaller _build_ closures, but do we want to?

Reply:
I'm still concerned about packages potentially retaining a reference to `$dev` though. With torch and jax moving towards JIT/AOT compilation, they just might do that...

Honestly, ideally we'd have a completely virtual output for `nix-support/propagated-build-inputs`, and arrows from it to all other inputs; having `dev` accommodate both headers and `nix-support/propagated-build-inputs` is very constraining